install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```
@@ -634,7 +634,7 @@ spec:
         resources:
           {{- toYaml . | trim | nindent 10 }}
         {{- end }}
-      {{- if and .Values.waitForKubeProxy (ne $kubeProxyReplacement "strict") }}
+      {{- if and .Values.waitForKubeProxy (or (ne $kubeProxyReplacement "strict") (ne $kubeProxyReplacement "true"))  }}
```

Let's say the user configured wait for kubepraxy and kubeproxyreplacement=strict. That will boil down into:

`and true (or (false) (true))`, so the result will be to wait for kube-proxy even though kube-proxy is disabled.

I think that the inner `or` needs to be an `and`...?

```suggestion
      {{- if and .Values.waitForKubeProxy (and (ne $kubeProxyReplacement "strict") (ne $kubeProxyReplacement "true"))  }}
```